From https://git.alpinelinux.org/cgit/aports/tree/main/qemu/0006-linux-user-signal.c-define-__SIGRTMIN-MAX-for-non-GN.patch

--- signal.c.old
+++ signal.c
@@ -26,6 +26,15 @@
 #include "target_signal.h"
 #include "trace.h"

+ //#define DEBUG_SIGNAL
+
+#ifndef __SIGRTMIN
+#define __SIGRTMIN 32
+#endif
+#ifndef __SIGRTMAX
+#define __SIGRTMAX (NSIG-1)
+#endif
+
 static struct target_sigaltstack target_sigaltstack_used = {
     .ss_sp = 0,
     .ss_size = 0,
